components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -969,9 +977,10 @@ public User updateUser(User user, Map<String, Boolean> requiredAttributes) throw
 
             // Get user claims mapped from SCIM dialect to WSO2 dialect.
             Map<String, String> claimValuesInLocalDialect = SCIMCommonUtils.convertSCIMtoLocalDialect(claims);
-            //If primary login identifire claim is enabled, pass that as a claim for userstoremanger.
+            //If primary login identifier claim is enabled, pass that as a claim for userstoremanger.
             if (isLoginIdentifiersEnabled() && StringUtils.isNotBlank(getPrimaryLoginIdentifierClaim())) {
-                claimValuesInLocalDialect.put(getPrimaryLoginIdentifierClaim(), user.getUsername());
+                claimValuesInLocalDialect.put(getPrimaryLoginIdentifierClaim(),
```

Comment:
can claimValuesInLocalDialect become null here? if so, shall we handle null check

Reply:
claimValuesInLocalDialect will not take a null value